UnityProject/Assets/Scripts/UI/Systems/Lobby/CharacterCustomization.cs

```diff
@@ -472,66 +491,81 @@ public void SetUpBodyPart(BodyPart bodyPart, bool InstantiateCustomisations = tr
 
 				if (bodyPart.OptionalOrgans.Count > 0)
 				{
-					var Option = Instantiate(AdditionalOrgan, ScrollListBody.transform);
-					Option.SetUp(this, bodyPart, "");
-					OpenBodyCustomisation[bodyPart.name] = (Option);
+					var option = Instantiate(AdditionalOrgan, ScrollListBody.transform);
+					option.SetUp(this, bodyPart, "");
+					OpenBodyCustomisation[bodyPart.name] = (option);
 				}
 
 
 				if (bodyPart.OptionalReplacementOrgan.Count > 0)
 				{
-					var Option = Instantiate(ReplacementOrgan, ScrollListBody.transform);
-					Option.SetUp(this, bodyPart, "");
-					OpenBodyCustomisation[bodyPart.name] = (Option);
+					var option = Instantiate(ReplacementOrgan, ScrollListBody.transform);
+					option.SetUp(this, bodyPart, "");
+					OpenBodyCustomisation[bodyPart.name] = (option);
 				}
 
 
 				//Setup sprite//
 				//OpenBodySprites
-				if (bodyPart?.OrganStorage?.Populater?.DeprecatedContents != null)
+				if (bodyPart != null && bodyPart.OrganStorage?.Populater?.DeprecatedContents != null)
 				{
-					foreach (var Organ in bodyPart.OrganStorage.Populater.DeprecatedContents)
+					foreach (var organ in bodyPart.OrganStorage.Populater.DeprecatedContents)
 					{
-						var subBodyPart = Organ.GetComponent<BodyPart>();
+						if (organ == null)
+						{
+							Logger.LogError($"[CharacterCustomization/SetUpBodyPart/Setup Sprites] - " + "Organ was detected as null!");
+							continue;
+						}
+						if (organ.TryGetComponent<BodyPart>(out var subBodyPart) == false) return;
 						ParentDictionary[bodyPart].Add(subBodyPart);
 						SetUpBodyPart(subBodyPart);
 					}
 				}
 			}
 		}
 
-		public void SetupBodyPartsSprites(BodyPart bodyPart)
+		private void SetupBodyPartsSprites(BodyPart bodyPart)
 		{
-			OpenBodySprites[bodyPart] = new List<SpriteHandlerNorder>();
-			var Sprites = bodyPart.GetBodyTypeSprites(ThisBodyType.bodyType); //Get the correct one
-
+			if (ThisBodyType == null || bodyPart == null)
+			{
+				Logger.LogError("[CharacterCustomization/SetupBodyPartSprites] - Unable to find a body! Are you sure you got one setup?");
+				return;
+			}
+			Tuple<SpriteOrder, List<SpriteDataSO>> Sprites = null;
 
-			if (Sprites != null)
+			try
 			{
-				if (Sprites?.Item1?.Orders == null || Sprites.Item1.Orders.Count == 0)
-				{
-					Logger.LogError("Rendering order not specified on " + bodyPart.name, Category.Character);
-				}
+				Sprites = bodyPart.GetBodyTypeSprites(ThisBodyType.bodyType); //Get the correct one
+			}
+			catch (Exception e)
```

Comment:
it is always best to catch the specific Exception you are expecting that could happen here.

Reply:
Shouldn't a log be enough?